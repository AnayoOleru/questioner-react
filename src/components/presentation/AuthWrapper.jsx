import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { get } from '../../utils/Storage';

class AuthWrapper extends Component {
  componentDidMount() {
    this.redirect();
  }

  redirect = () => {
    const { history } = this.props;
    const token = get('token');
    if (!token) {
      return history.push('/login');
    }
  };

  render() {
    const { children } = this.props;
    return children;
  }
}

export default withRouter(AuthWrapper);
