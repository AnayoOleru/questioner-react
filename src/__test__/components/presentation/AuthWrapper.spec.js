import React from 'react';
import { shallow } from 'enzyme';
import AuthWrapper from '../../../components/presentation/AuthWrapper';

describe('Auth Wrapper Component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<AuthWrapper />);
  });
  test('Auth Wrapper component should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
