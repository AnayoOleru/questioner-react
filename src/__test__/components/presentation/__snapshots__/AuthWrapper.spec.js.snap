// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Wrapper Component Auth Wrapper component should render 1`] = `ShallowWrapper {}`;
